mkdocs_autolinks_plugin/plugin.py

```diff
@@ -2,7 +2,15 @@
 import os
 from mkdocs.plugins import BasePlugin
 
-AUTOLINK_RE = r'\[([^\]]+)\]\(([^)/]+\.(md|png|jpg))\)'
+# For Regex, match groups are:
+#       0: Whole markdown link e.g. [Alt-text](url)
+#       1: Alt text
+#       2: Full URL e.g. url + hash anchor
+#       3: Filename e.g. filename.md
+#       4: File extension e.g. .md, .png, etc.
+#       5. hash anchor e.g. #my-sub-heading-link
+
+AUTOLINK_RE = r'\[([^\]]+)\]\((([^)/]+\.(md|png|jpg))(#.*)*)\)'
```

Comment:
The hash anchor group matches eagerly, which causes all kinds of errors, since it will match past the actual end parenthesis of the link if it can. It should be changed to `(#[^)]*)`, making the final line
```python
AUTOLINK_RE = r'\[([^\]]+)\]\((([^)/]+\.(md|png|jpg))(#[^)]*)*)\)'
```